test/unit/test_reduce_by_key.cu

```diff
@@ -87,7 +85,7 @@ TEST(reduce_by_key, single_repeated_index)
 TEST(reduce_by_key, scatter)
 {
     std::vector<int> index = {0,0,0,1,2,2,2,2,3,3,7,7,7,7,7,11};
-    unsigned n = util::max_value(index)+1;
+    unsigned n = index.back()+1;
```

Comment:
It does not require monotonicity, though we have never tested that, because in the library it is always applied to a monotonic sequence.
Using `std::maximum_element` makes the test harder to understand (`max_value` and `::back` aren't clear either), so I will hard code the value `n=12` and add a comment explaining the test.